Common/Scheduling/TimeRules.cs

```diff
@@ -215,5 +229,29 @@ private static IEnumerable<DateTime> EveryIntervalIterator(IEnumerable<DateTime>
                 }
             }
         }
+
+        /// <summary>
+        /// Check whether the data resolution is compatible with a interval defined by a <see cref="ITimeRule"/>. 
+        /// </summary>
+        /// <param name="name"><see cref="ITimeRule"/> name</param>
+        /// <param name="interval">The interval defined by a <see cref="ITimeRule"/></param>
+        /// <param name="security">The security with the data subscriptions. If not defined, will use all the available securities</param>
+        public void ValidateInterval(string name, TimeSpan interval, Security security = null)
+        {
+            var resolution = (security?.Subscriptions ?? _securities.SelectMany(x => x.Value.Subscriptions))
+                .GetHighestResolution();
+
+            if (resolution == Resolution.Daily && interval != TimeSpan.Zero ||
+                resolution == Resolution.Hour && interval.TotalHours - interval.Hours > 0)
+            {
+                Logging.Log.Trace(
```

Comment:
Please make this reach the user, not just the logs.

Reply:
Interesting! As time goes on, the number of components that need access to logging functionality increases. The running solution has been to throw `IAlgorithm` references around. Perhaps it's time to extract the `IAlgorithm`'s logging functions out into a separate interface and have `IAlgorithm` implement that new interface. In this way, we can avoid injecting the fattest abstract in LEAN (`IAlgorithm`) into one of the smallest components.